sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -611,7 +611,7 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
 
   def functionWithUnsupportedSyntaxError(prettyName: String, syntax: String): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1023",
+      errorClass = "INVALID_SQL_SYNTAX.FUNCTION_WITH_UNSUPPORTED_SYNTAX",
       messageParameters = Map("prettyName" -> prettyName, "syntax" -> syntax))
```

Comment:
Please, quote `prettyName`:
```suggestion
      messageParameters = Map("prettyName" -> toSQLId(prettyName), "syntax" -> syntax))
```

Reply:
And wrap `syntax` by `toSQLStmt`

Reply:
HaHa! I forgot it.